File: 1548-check-if-all-1s-are-at-least-length-k-places-away/1548-check-if-all-1s-are-at-least-length-k-places-away.c
bool kLengthApart(int* nums, int numsSize, int k) {
    int i,f=0,c=0;
 for(i=0;i<numsSize;i++)
 {
    if(nums[i]==0)
    {
    if(f==1)
    c++;
    }
   else if(nums[i] == 1)
    {
    if(f==1 && c<k)
    return false;
     f=1;
     c=0;
    }
}
return true;   
}

File: 1548-check-if-all-1s-are-at-least-length-k-places-away/1548-check-if-all-1s-are-at-least-length-k-places-away.java
class Solution {
    public boolean kLengthApart(int[] nums, int k) {
        int s=0,f=0;;
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]==1)
            {
            if(s<k && f==1)
            return false;
            f=1;
            s=0;
            }
            else
            s++;
        }
        return true;
    }
}